### PR TITLE
Bv 105/redux actions improve logic

### PR DESCRIPTION
**Pull request by wojciech-nowakowski**

### CODE REVIEW

File: src/store/actions/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import * as ethereumWalletActions from './ethereum/wallets';
+import * as tezosWalletActions from './tezos/wallets';
+
+
+export const getWalletByDatasource = (
+    dataSource: string
+   ): any => {
+    switch (dataSource) {
+      case 'TEZOS':
```
Comment (Milena-Wasiak) [inline src/store/actions/helpers.ts]: Może nazwy blockchainów przeniesiemy do enumów, unikniemy literówek i nazwę ewentualnie zmieni się w jednym miesjcu

**Review by Milena-Wasiak — APPROVED** [review]

File: src/components/BarChart/BarChart-container.tsx [hunk 2]
```diff
@@ -119,8 +119,7 @@ const BarChartContainer = (props: BarChartProps) => {
   }, [walletHash]);
 
   useEffect((): void => {
-    actions.fetchEthereumWallets();
-    actions.fetchTezosWallets();
+    actions.fetchWalletsByDataSource(dataSource);
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-container.tsx]: fetchWalletsByDataSource -> fetchWalletsByBlockchain
is more descriptive

File: src/components/BarChart/BarChart-redux.tsx [hunk 3]
```diff
@@ -4,47 +4,41 @@ import { useMappedState, useDispatch } from 'redux-react-hook';
 import BarChartContainer from './BarChart-container';
 import { State, Customization } from './types';
 import { Wallets } from '../../types';
+import { getBlockchainByDatasource } from '../../store/reducers/helpers';
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-redux.tsx]: selectors should always be in the same file as their reducers.

File: src/components/DashboardFilters/DashboardFilters-container.tsx [hunk 4]
```diff
@@ -139,9 +139,8 @@ const Filters = (props: any) => {
 
   const fetchNewData = (activeBlockchains: string[]) => {
     activeBlockchains.forEach((blockchain: string) => {
-      console.log(filters);
```
Comment (damianlewandowski) [inline src/components/DashboardFilters/DashboardFilters-container.tsx]: Why not just call `actions.fetchWalletsByDataSource` with parameter `blockchain` like `actions.fetchWalletsByDataSource({limit: filters.limit, from: filters.from, to: filters.to}, blockchain)`?

You should never NOT use variables in cases where the same value is used in multiple places, unless you want to get intimate with buggy and bad code.

Declare some const enums or variables e.g. 
```
enum Blockchains {
  ETH = 'Ethereum',
  TZX = 'Tezos'
}
```
and use them wherever you can.

File: src/store/reducers/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export const getBlockchainByDatasource = (
```
Comment (damianlewandowski) [inline src/store/reducers/helpers.ts]: Create a file `blockchainSelectors.ts` and paste this code in there. `helpers.ts` is too ambiguous.

**Review by damianlewandowski — CHANGES_REQUESTED** [review]

**Comment by damianlewandowski** [comment]
Good idea with using selector here 👍

File: src/components/BarChart/BarChart-container.tsx (hunk 2 above)
Comment (wojciech-nowakowski) [reply src/components/BarChart/BarChart-container.tsx]: fixed

**Pull request merged**